src/weights.jl

```diff
@@ -1,9 +1,9 @@
 
 ###### Weight vector #####
 
-immutable WeightVec{W,Vec<:RealVector}
-    values::Vec
-    sum::W
+immutable WeightVec{T, V<:RealVector, S} <: RealVector{T}
```

Comment:
@nalimilan Isn't `WeightVec{S<:Real, T<:Real, V<:AbstractVector{T}} <: AbstractVector{T}` triangular dispatch which is only supported in 0.6?